cmd/snap/last.go

```diff
@@ -0,0 +1,85 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+)
+
+type lastMixin struct {
+	LastChangeType string `long:"last"`
+	Positional     struct {
+		ID changeID `positional-arg-name:"<id>"`
+	} `positional-args:"yes"`
+}
+
+var lastMixinOptDesc = map[string]string{
+	"last": i18n.G("Select last change of given type (install, refresh, remove, try, auto-refresh etc.)"),
+}
+
+var lastMixinArgDesc = []argDesc{{
+	name: i18n.G("<change-id>"),
+	desc: i18n.G("Change ID"),
+}}
+
+func (l *lastMixin) GetChangeID(cli *client.Client) (changeID, error) {
```

Comment:
seems it would need less casts if this returned string and showChange took also a string